ql/cashflows/cashflows.cpp

```diff
@@ -625,9 +625,18 @@ namespace QuantLib {
                     }
                     refEndDate = couponDate;
                 }
-
-                t += dc.yearFraction(lastDate, couponDate,
-                                     refStartDate, refEndDate);
+                
+                if (coupon && lastDate!=refStartDate){
+                     Time couponPeriod = y.dayCounter().yearFraction(refStartDate,
+                                                couponDate, refStartDate, refEndDate);
+                     Time accruedPeriod = y.dayCounter().yearFraction(refStartDate,
+                                                lastDate, refStartDate, refEndDate);
+                     t += couponPeriod - accruedPeriod;                                
+                }
+                else{
+                     t += dc.yearFraction(lastDate, couponDate,
+                                 refStartDate, refEndDate);
+                }
```

Comment:
This calculation is repeated three times.  I'd put it into a helper function. (It can go into this file, in an anonymous namespace).